### PR TITLE
Sonar

### PR DESCRIPTION
**Pull request by ardangor**
Please review my changes

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -13,6 +13,7 @@ public class TorpedoStore {
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  private Random rand = new Random();
```
Comment (Dolnay-Adam) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:16]: Security issues

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (Dolnay-Adam) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: Glorious

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
+    double r = this.rand.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (Dolnay-Adam) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: Great catch

**Review by Dolnay-Adam — APPROVED** [review]

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (boglarka-bajkai) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:16]: I like this new change

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 2 above)
Comment (boglarka-bajkai) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: Yeet it

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 3 above)
Comment (boglarka-bajkai) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: Yay

**Review by boglarka-bajkai — CHANGES_REQUESTED** [review]
Hi

**Pull request merged**